projects/kit/components/combo-box/combo-box.component.ts

```diff
@@ -183,9 +183,8 @@ export class TuiComboBoxComponent<T>
 
         const match = this.accessor?.getOptions().find(item => this.isStrictMatch(item));
 
-        if (match !== undefined) {
+        if (match) {
             this.updateValue(match);
-            this.updateSearch(null);
```

Comment:
When a user picks value from the list we emit null for search so that it's clear no need to fetch serverside items for the new string value. So that's correct. Something else got broken, because it used to work properly this way.

Reply:
revert